include/pisa/wand_data.hpp

```diff
@@ -25,72 +28,98 @@ class wand_data {
     wand_data(LengthsIterator len_it,
               uint64_t num_docs,
               binary_freq_collection const &coll,
+              std::string const &scorer_name,
               BlockSize block_size)
     {
         std::vector<uint32_t> doc_lens(num_docs);
+        uint32_t max_len = 0;
         std::vector<float> max_term_weight;
-        std::vector<uint32_t> terms_count;
+        std::vector<uint32_t> term_counts;
+        std::vector<uint32_t> term_lens;
         global_parameters params;
         double collection_len = 0;
         spdlog::info("Reading sizes...");
 
         for (size_t i = 0; i < num_docs; ++i) {
-            float len = *len_it++;
+            uint32_t len = *len_it++;
+            max_len = std::max(max_len, len);
             doc_lens[i] = len;
             collection_len += len;
         }
 
         float avg_len = float(collection_len / double(num_docs));
 
         typename block_wand_type::builder builder(coll, params);
+
+        for (auto const &seq : coll) {
+            size_t term_count = std::accumulate(seq.freqs.begin(), seq.freqs.end(), 0);
+            term_counts.push_back(term_count);
```

Comment:
The naming is confusing, you're not counting terms but occurrences, right? Maybe `term_occurrence_counts` would be a better name? Same with `term_lens`---these are just term frequencies, right?

Reply:
Agreed on `term_occurrence_counts`.
`term_lens` are the lengths of the posting list.  It is the number of postings per term. Do you prefer `term_posting_counts`?